src/components/Admin/AddPopUpClinic/AdminPopUpClinic.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react";
+import { Card, Form, FormLayout, TextField } from "@shopify/polaris";
+import { useTranslation } from "react-i18next";
+import { postalCodeIsValid } from "../../../utils/postalCode";
+import { phoneNumberIsValid } from "../../../utils/phoneNumber";
+
+export interface IPopUpClinic {
+  name: string;
+  date: string;
+  street: string;
+  city: string;
+  province: string;
+  postalCode: string;
+  phone: string;
+  websiteUrl: string;
+  [key: string]: string;
+}
+
+const handleSubmit = () => {
+  return true;
+};
+
+export function AdminPopUpClinic() {
+  const { t } = useTranslation();
+
+  const [popUpClinic, setPopUpClinc] = useState<IPopUpClinic>({
+    name: "",
+    date: new Date().toDateString(),
+    street: "",
+    city: "",
+    province: "",
+    postalCode: "",
+    phone: "",
+    websiteUrl: "",
+  });
+
+  const [popUpClinicErrors, setPopUpClincErrors] = useState<IPopUpClinic>({
+    name: "",
+    date: "",
+    street: "",
+    city: "",
+    province: "",
+    postalCode: "",
+    phone: "",
+    websiteUrl: "",
+  });
+
+  const onFormChanged = (field: string, value: string) => {
+    if (field === "postalCode") {
+      if (!postalCodeIsValid(value)) {
+        setPopUpClincErrors({
+          ...popUpClinic,
+          [field]: t("invalidpostalcode"),
+        });
+      } else {
+        setPopUpClincErrors({ ...popUpClinic, [field]: "" });
```

Comment:
We only have two business rules to validate? Are any of the other fields mandatory?